### PR TITLE
Left dependency Admin Tools Application (Pro) - Default causes license error #100

### PR DESCRIPTION
**Pull request by ChiuchiuSorin**
Created a listener that attempts the removal of the rest of Admin Tools modules.

### CODE REVIEW

File: application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                    logger.info("Successfully uninstalled all Admin Tools modules.");
+                }
+            } catch (UninstallException e) {
+                logger.error("There was an error while uninstalling Admin Tools modules.");
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private String getExtensionId(String module)
```
Comment (oanalavinia) [inline application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java:99]: `getAdminToolsExtensionId`. I know it's longer, but it was too generic
Maybe `getATExtensionId`? I know it's a common practice to use acronyms for extensions when the name it's too long

File: application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                        new ExtensionId(getExtensionId("api"), uninstalledExtension.getVersion());
+                    ExtensionId defaultExtensionId =
+                        new ExtensionId(getExtensionId("default"), uninstalledExtension.getVersion());
+
+                    InstalledExtension apiExtension = installedRepository.getInstalledExtension(apiExtensionId);
+                    InstalledExtension defaultExtension = installedRepository.getInstalledExtension(defaultExtensionId);
+                    String namespace = new WikiNamespace(wikiContextProvider.get().getWikiId()).serialize();
+                    installedRepository.uninstallExtension(defaultExtension, namespace);
+                    installedRepository.uninstallExtension(apiExtension, namespace);
+                    logger.info("Successfully uninstalled all Admin Tools modules.");
```
Comment (oanalavinia) [inline application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java:90]: I would add more debug logs besides this one (e.g. attempting to uninstall the api module,..)

File: application-admintools-ui/src/main/resources/AdminTools/Code/Configuration.xml [hunk 3]
```diff
@@ -105,7 +105,7 @@
       <serverLocation/>
     </property>
     <property>
-      <spamSize>2</spamSize>
+      <spamSize>50</spamSize>
```
Comment (oanalavinia) [inline application-admintools-ui/src/main/resources/AdminTools/Code/Configuration.xml:108]: Why is this `spamSize` field doing?

File: application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    /**
+     * Creates an event-listener filtering for ExtensionUninstalledEvent.
+     */
+    public AdminToolsUninstallListener()
+    {
+        super(HINT, new ExtensionUninstalledEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
```
Comment (oanalavinia) [inline application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java:74]: Could you add some unit tests for this method?

**Review by oanalavinia — CHANGES_REQUESTED** [review]

File: application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java (hunk 1 above)
Comment (ChiuchiuSorin) [reply application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsUninstallListener.java:99]: I think `getAdminToolsExtensionId` it's fine. It does not break any line and it's clear.

File: application-admintools-ui/src/main/resources/AdminTools/Code/Configuration.xml (hunk 3 above)
Comment (ChiuchiuSorin) [reply application-admintools-ui/src/main/resources/AdminTools/Code/Configuration.xml:108]: It's a leftover from testing that was released. I was using a comments spam size of 2, but the default should be at 50.

**Review by oanalavinia — APPROVED** [review]

**Pull request merged**